controllers/provisioning_controller.go

```diff
@@ -185,11 +186,17 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return ctrl.Result{}, err
 	}
 
-	if err := provisioning.ValidateBaremetalProvisioningConfig(baremetalConfig); err != nil {
+	if errs := provisioning.ValidateBaremetalProvisioningConfig(baremetalConfig); len(errs) > 0 {
 		// Provisioning configuration is not valid.
 		// Requeue request.
-		r.Log.Error(err, "invalid config in Provisioning CR")
-		err = r.updateCOStatus(ReasonInvalidConfiguration, err.Error(), "Unable to apply Provisioning CR: invalid configuration")
+		var errorstring string
+		for _, err := range errs {
+			errorstring = err.Error() + "\n" + errorstring
+		}
+		errorstring = strings.TrimSuffix(errorstring, "\n")
+
+		r.Log.Error(errors.New(errorstring), "invalid config in Provisioning CR")
```

Comment:
This is going to produce a multi-line log message. What do you think about joining the strings on `"; "` instead of newlines?

Reply:
Yeah either way.  Maybe that makes more sense.

Reply:
I changed it to "; "